src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Application.cs

```diff
@@ -990,10 +997,17 @@ public ThemeMode ThemeMode
 
                 if(!_resourcesInitialized)
                 {
-                    // If the resources are not initializd, 
-                    // fluent dictionary included will be reset.
-                    // Hence, deferring the step.
-                    ThemeManager.DeferredAppThemeLoading = true;
+                    // We do this because, Resources may not be present in App.xaml
+                    // so the setter will never be called, and we will not get a chance to 
+                    // load the fluent dictionary before parsing Window, where Fluent resources
+                    // may be used.
+                    ThemeManager.OnApplicationThemeChanged(oldValue, value);
+                    _resourcesInitialized = false;
```

Comment:
Is this needed? We are here only if `_resourcesInitialized` is already `false`.

Reply:
Yes, because OnApplicatonThemeChanged will set the _resourcesInitialized to true